src/chainparams.h

```diff
@@ -107,7 +108,9 @@ class CChainParams
  * @returns a CChainParams* of the chosen chain.
  * @throws a std::runtime_error if the chain is not supported.
  */
-std::unique_ptr<CChainParams> CreateChainParams(const std::string& chain);
+std::unique_ptr<const CChainParams> CreateChainParams(const std::string& chain);
+/** Extended version taking custom args (only useful for regtest). */
+std::unique_ptr<const CChainParams> CreateChainParams(const std::string& chain, ArgsManager& args);
```

Comment:
I tend to agree with @ajtowns that we can do without `std::unique_ptr<const CChainParams> CreateChainParams(const std::string& chain, ArgsManager& args)`.

If you want to keep it then maybe you can remove its declaration from `chainparams.h`. There is no need to expose it; it only uses internal linkage at the moment.